share/spack/templates/container/Dockerfile

```diff
@@ -34,10 +41,10 @@ COPY --from=builder {{ paths.store }} {{ paths.store }}
 COPY --from=builder {{ paths.view }} {{ paths.view }}
 COPY --from=builder /etc/profile.d/z10_spack_environment.sh /etc/profile.d/z10_spack_environment.sh
 
-{% if os_packages %}
-RUN {{ os_packages.update }}                                   \
- && {{ os_packages.install }}{% for pkg in os_packages.list %} {{ pkg }}{% endfor %} \
- && {{ os_packages.clean }}
+{% if os_packages_final %}
+RUN {{ os_packages_final.update }}                                   \
+ && {{ os_packages_final.install }}{% for pkg in os_packages_final.list %} {{ pkg }}{% endfor %} \
```

Comment:
```suggestion
RUN {{ os_packages_final.update }} \
 && {{ os_packages_final.install }} {{ os_packages_final.list | join | replace('\n', ' ') }} \
```

Reply:
That would work too. Is there any reason to prefer this over what is in the PR right now?

Reply:
Consistency with the singularity recipe maybe?

Clarity maybe? The main point is "join the elements with a space". The space in the for loop is hidden nicely between a closing and an opening brace. The `|join…` shows that in a cleaner form.

Reply:
Well, these seems equivalent to me. Since I have no preference change done in 0baf4d8